src/oca_github_bot/commands.py

```diff
@@ -51,11 +58,30 @@ def __init__(self, name, options):
         self.parse_options(options)
 
     @classmethod
-    def create(cls, name, options):
-        if name == "merge":
-            return BotCommandMerge(name, options)
-        else:
-            raise InvalidCommandError(name)
+    def create(cls, name, options, github_context):
+        try:
+            if name == "merge":
+                return BotCommandMerge(name, options)
+            else:
+                raise InvalidCommandError(name)
+        except CommandError as e:
```

Comment:
I like the CommandError exception hierarchy.

However I would rather move this error handling to the webhook, so the BotComand class need to be concerned about the context in which it was called (could be PR but also an issue, and why not something else such a CLI in the future).

Reply:
Thanks for your advice. I moved in the webhook. Indeed, I was not very confortable with the current design.
Furthermore, I had to add a context=None, in the end of some function, and it reminded me of the dark hours of openerp. ;-)
Could you check if it's ok now ?